src/autoscaler/models/custom_metrics.go

```diff
@@ -14,7 +14,29 @@ type MetricsConsumer struct {
 	CustomMetrics []*CustomMetric `json:"metrics"`
 }
 
-type CustomMetricCredentials struct {
-	Username string
-	Password string
+type Credential struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+//custom metrics response
+//  {
+//     credentials: {
+//         custom_metrics:{
+//           username: username,
+//           password: password,
+//           url: settings.customMetricsUrl
+//         }
+//     }
+// }
+
+type CustomMetrics struct {
```

Comment:
please rename to CustomMetricsCredentials , since there is another quite similar struct definition of "customemetric" in L3